src/init/common.cpp

```diff
@@ -78,27 +79,24 @@ void SetLoggingLevel(const ArgsManager& args)
     }
 }
 
-void SetLoggingCategories(const ArgsManager& args)
+static constexpr std::array DEBUG_LOG_OPTS{"-debug", "-debugexclude"};
+
+static void EnableOrDisableLogCategories(const ArgsManager& args, const std::string& opt)
 {
-    if (args.IsArgSet("-debug")) {
-        // Special-case: if -debug=0/-nodebug is set, turn off debugging messages
-        const std::vector<std::string> categories = args.GetArgs("-debug");
-
-        if (std::none_of(categories.begin(), categories.end(),
-            [](std::string cat){return cat == "0" || cat == "none";})) {
-            for (const auto& cat : categories) {
-                if (!LogInstance().EnableCategory(cat)) {
-                    InitWarning(strprintf(_("Unsupported logging category %s=%s."), "-debug", cat));
-                }
-            }
-        }
+    Assume(std::any_of(DEBUG_LOG_OPTS.cbegin(), DEBUG_LOG_OPTS.cend(), [&](const auto& d) { return opt == d; }));
+    if (!args.IsArgSet(opt)) return;
+    const std::vector<std::string>& categories{args.GetArgs(opt)};
+    if (std::any_of(categories.cbegin(), categories.cend(), [](const auto& c) { return LogInstance().IsNoneCategory(c); })) return;
+    for (const auto& c : categories) {
+        const bool success{opt == DEBUG_LOG_OPTS[0] ? LogInstance().EnableCategory(c) : LogInstance().DisableCategory(c)};
```

Comment:
`DEBUG_LOG_OPTS[0] `

Feels like this is undoing whatever convenience/readability you set up by defining `DEBUG_LOG_OPTS` in the first place. Do you think more debug args will be added in the future? Or can we just use

`args.IsArgSet("-debug") || args.IsArgSet("-debugexclude")` ... etc

Reply:
Well I meant something like a switch statement with `"-debug"` and `"-debugexclude"` literal strings as cases. I think that would make it more readable and if anything is ever added to `DEBUG_LOG_OPTS[]` it won't necessarily break. But that might just be too long or out of style

Reply:
Yeah 😬 waddaya think?